task/jib-gradle/0.4/jib-gradle.yaml

```diff
@@ -0,0 +1,110 @@
+  - name: IMAGE
+    description: Reference of the image gradle will produce
+  - name: DIRECTORY
+    description: The directory containing the app, relative to the source repository root
+    default: .
+  - name: CACHE
+    description: The name of the volume for caching Gradle artifacts, local Maven repository, and base image layers
+    default: empty-dir-volume
+  - name: INSECUREREGISTRY
+    description: Whether to allow insecure registry
+    default: "false"
+  - name: EXTRA_ARGS
+    description: Extra arguments to add to the gradle jib build
+    default: ""
+
+  workspaces:
+  - name: source
+
+  results:
+  - name: IMAGE_DIGEST
+    description: Digest of the image just built.
+  - name: IMAGE_URL
+    description: URL of the image just built.
+
+  steps:
+  - name: build-and-push
+    image: $(params.BUILDER_IMAGE)
+    workingDir: $(workspaces.source.path)/$(params.DIRECTORY)
+    script: |
+      #!/bin/sh
+      set -o errexit
+
+      # Adds Gradle init script that applies the Jib Gradle plugin.
+      echo "initscript {
+              repositories { maven { url 'https://plugins.gradle.org/m2' } }
+              dependencies { classpath 'gradle.plugin.com.google.cloud.tools:jib-gradle-plugin:+' }
+            }
+            rootProject {
+              afterEvaluate {
+                if (!project.plugins.hasPlugin('com.google.cloud.tools.jib')) {
+                  project.apply plugin: com.google.cloud.tools.jib.gradle.JibPlugin
+                }
+              }
+            }" > "$HOME"/init-script.gradle
+
+      # Runs the Gradle Jib build.
+      gradle jib \
+        --stacktrace --console=plain \
+        --init-script="$HOME"/init-script.gradle \
+        --gradle-user-home="$HOME"/.gradle \
+        -Dgradle.user.home="$HOME"/.gradle \
+        -Duser.home="$HOME" \
+        -Djib.allowInsecureRegistries="$(params.INSECUREREGISTRY)" \
+        -Djib.to.image="$(params.IMAGE)" \
+        -Djib.outputPaths.digest="$(results.IMAGE_DIGEST.path)" \
+        $(params.EXTRA_ARGS)
+    env:
+    - name: HOME
+      value: /workspace
+    - name: "DOCKER_CONFIG"
+      value: $(credentials.path)/.docker/
+    volumeMounts:
+    # empty volume required to make the Gradle home globally writable
+    - name: empty-dir-volume
+      mountPath: /workspace/.gradle
+      subPath: gradle-user-home
+    - name: $(params.CACHE)
+      mountPath: /workspace/.gradle/caches
+      subPath: gradle-caches
+    - name: $(params.CACHE)
+      mountPath: /workspace/.gradle/wrapper
+      subPath: gradle-wrapper
+    - name: $(params.CACHE)
+      mountPath: /workspace/.m2
+      subPath: m2-cache
+    - name: $(params.CACHE)
+      mountPath: /workspace/.cache
+      subPath: jib-cache
+    securityContext:
+      runAsUser: 0
+
+  - name: write-url
+    image: $(params.BUILDER_IMAGE)
+    script: echo $(params.IMAGE) | tee $(results.IMAGE_URL.path)
```

Comment:
can we merge this step with the previous one as both the steps are using the same image and in step 1 we are already writing the script so maybe we can merge these. WDYT?